at_commons/lib/src/keystore/at_key.dart

```diff
@@ -357,9 +395,12 @@ class Metadata {
   /// Represents the type of encoding (ex: base64) when the value contains a new line character's
   String? encoding;
 
+  /// When set to true, implies the key is [LocalKey]
+  bool isLocal = false;
```

Comment:
Just thinking about this some more, it is inconsistent to have this instance variable be part of Metadata ... it should be part of AtKey. Then we can add a getter and setter for AtKey.isLocal, and have setter only allow setting isLocal to true when sharedWith is null. Likewise we can also extend the setter for sharedWith so that it only allows setting of sharedWith when isLocal is false.

Reply:
@gkc: Shall i go about moving isLocal to AtKey class from Metadata?